lib/postgres.js

```diff
@@ -47,14 +47,100 @@ class Postgres {
             stickers    jsonb,
             updated     timestamp NOT NULL,
             rarity      smallint NOT NULL,
+            floatid     bigint  NOT NULL,
             PRIMARY KEY (a)
-        )`);
+        );
+
+        CREATE TABLE IF NOT EXISTS history (
+            floatid     bigint  NOT NULL,
+            a           bigint  NOT NULL,
+            steamid     bigint  NOT NULL,
+            created_at  timestamp NOT NULL,
+            PRIMARY KEY (floatid, a)
+        );
+
+        ALTER TABLE items ADD COLUMN IF NOT EXISTS floatid BIGINT;
+
+        -- Float ID is defined as the first asset id we've seen for an item
+
+        CREATE OR REPLACE FUNCTION is_steamid(IN val bigint)
+            RETURNS boolean
+            LANGUAGE 'plpgsql'
+            PARALLEL SAFE
+        AS $BODY$BEGIN
+            IF val < 76561197960265728 THEN
+                RETURN FALSE;
+            ELSIF (val >> 56) > 5 THEN
+                RETURN FALSE;
+            ELSIF ((val >> 32) & ((1 << 20) - 1)) > 32 THEN
+                RETURN FALSE;
+            END IF;
+        
+            RETURN TRUE;
+        END;$BODY$;
+        
+        CREATE OR REPLACE FUNCTION update_history()
+        RETURNS TRIGGER
+        AS $$
+        DECLARE
+            floatid BIGINT;
+        BEGIN
+            -- Handle cases where the floatid isn't there
+            IF NEW.floatid IS NOT NULL THEN
+                floatid = NEW.floatid;
+            ELSE
+                -- Update the DB with the floatid if it still doesn't exist
+                UPDATE items SET floatid=OLD.a WHERE a=NEW.a;
```

Comment:
Wouldn't this cause a loop with the trigger for update_history?

Reply:
It doesn't, but isn't required if we do BEFORE UPDATE, so I've modified this.